app/model/jobs/steps/ModifyContentTags.scala

```diff
@@ -0,0 +1,70 @@
+package model.jobs.steps
+
+import com.gu.tagmanagement.{OperationType, TagWithSection, TaggingOperation}
+import model.{BatchTagOperation, Section, Tag, TagAudit}
+import model.jobs.StepStatus
+
+import scala.concurrent.duration._
+import model.jobs.Step
+import play.api.Logger
+import services.KinesisStreams
+import repositories.{ContentAPI, TagAuditRepository}
+
+import scala.language.postfixOps
+
+// The 'op' argument of this case class should really be the BatchTagOperation type but serializing enums to JSON isn't very easy with the
+// current version of play and enumeratim
+case class ModifyContentTags(tag: Tag, section: Option[Section], contentIds: List[String], op: String, `type`: String = ModifyContentTags.`type`, var stepStatus: String = StepStatus.ready, var stepMessage: String = "Waiting", var attempts: Int = 0) extends Step {
+  override def process = {
+    contentIds foreach { contentPath =>
+      val taggingOperation = TaggingOperation(
+        operation = OperationType.valueOf(op.replace("-", "")).get,
+        contentPath = contentPath,
+        tag = Some(TagWithSection(tag.asThrift, section.map(_.asThrift)))
+      )
+      KinesisStreams.taggingOperationsStream.publishUpdate(contentPath.take(128), taggingOperation)
+      Logger.info(s"raising $op for ${tag.id} on $contentPath operation")
+    }
+    TagAuditRepository.upsertTagAudit(TagAudit.batchTag(tag, op, contentIds.length))
+  }
+
+  override def waitDuration: Option[Duration] = {
+    Some(5 seconds)
+  }
+
+  override def check: Boolean = {
+    val count = ContentAPI.countOccurencesOfTagInContents(contentIds, tag.path)
+
+    // As stated above, once we get the BatchTagOperation enum type serializable we can match against a type here instead of a string
+    val expected = op match {
+      case "remove" => 0
+      case _ => contentIds.length
+    }
+
+    Logger.info(s"Checking batch tag operations. Expected=$expected Actual=$count")
+    if (count == expected) {
```

Comment:
This can simply be written as `count == expected`.

Reply:
We had a discussion about what was clearer here. I personally find Scala's lack of `return` statements and semicolons can make a tail-end boolean evaluation a tad confusing. Will change though.